rustls/src/crypto/ring/kx.rs

```diff
@@ -76,22 +91,38 @@ pub static X25519: &dyn SupportedKxGroup = &KxGroup {
     //  are not compliant to SP 800-56Arev3."
     // -- <https://csrc.nist.gov/csrc/media/Projects/cryptographic-module-validation-program/documents/fips%20140-3/FIPS%20140-3%20IG.pdf>
     fips_allowed: false,
+
+    pub_key_validator: x25519_validator,
 };
 
+fn x25519_validator(point: &[u8]) -> bool {
+    point.len() == 32
+}
+
 /// Ephemeral ECDH on secp256r1 (aka NIST-P256)
 pub static SECP256R1: &dyn SupportedKxGroup = &KxGroup {
     name: NamedGroup::secp256r1,
     agreement_algorithm: &agreement::ECDH_P256,
     fips_allowed: true,
+    pub_key_validator: secp256r1_validator,
 };
 
+fn secp256r1_validator(point: &[u8]) -> bool {
+    matches!((point.len(), point.first()), (65, Some(&0x04)))
```

Comment:
Citation for the `0x04` constant? (I also think Djc's feedback to drop the `&` could be applied here).

If you agree, ditto for L123.

Reply:
Added a citation of the 0x04 number, and also refactored this to remove the length checks (there's actually no need for that).